crates/bls-crypto/src/hash_to_curve/try_and_increment.rs

```diff
@@ -74,98 +81,130 @@ where
 impl<'a, H, P> TryAndIncrement<'a, H, P>
 where
     H: XOF<Error = BLSError>,
-    P: Bls12Parameters,
+    P: SWModelParameters,
 {
+    /// Hash with attempt takes the input, appends a counter
     pub fn hash_with_attempt(
         &self,
         domain: &[u8],
         message: &[u8],
         extra_data: &[u8],
-    ) -> Result<(G1Projective<P>, usize), BLSError> {
-        const NUM_TRIES: usize = 256;
-        const EXPECTED_TOTAL_BITS: usize = 512;
-        const LAST_BYTE_MASK: u8 = 1;
-        const GREATEST_MASK: u8 = 2;
-
-        let fp_bits =
-            (((<P::Fp as PrimeField>::Params::MODULUS_BITS as f64) / 8.0).ceil() as usize) * 8;
-        let num_bits = fp_bits;
-        let num_bytes = num_bits / 8;
-
-        //round up to a multiple of 8
-        let hash_fp_bits =
-            (((<P::Fp as PrimeField>::Params::MODULUS_BITS as f64) / 256.0).ceil() as usize) * 256;
-        let hash_num_bits = hash_fp_bits;
-        assert_eq!(hash_num_bits, EXPECTED_TOTAL_BITS);
-        let hash_num_bytes = hash_num_bits / 8;
-        let mut counter: [u8; 1] = [0; 1];
+    ) -> Result<(GroupProjective<P>, usize), BLSError> {
+        let num_bytes = GroupAffine::<P>::SERIALIZED_SIZE;
+        // TODO: How can we properly find the extension?
+        const BASE_SIZE: usize = 48;
```

Comment:
True only for bls12-377/381, and yeah, would be great to have a proper method for the extension size